tests/bdd-smoke/utils/libvirt_inst_state_file_validator.go

```diff
@@ -0,0 +1,160 @@
+			SSHKey     string `json:"sshKey"`
+			BaseDomain string `json:"baseDomain"`
+			Networking struct {
+				Type            string `json:"type"`
+				ServiceCIDR     string `json:"serviceCIDR"`
+				ClusterNetworks []struct {
+					Cidr             string `json:"cidr"`
+					HostSubnetLength int    `json:"hostSubnetLength"`
+				} `json:"clusterNetworks"`
+			} `json:"networking"`
+			Machines []struct {
+				Name     string `json:"name"`
+				Replicas int    `json:"replicas"`
+				Platform struct {
+				} `json:"platform"`
+			} `json:"machines"`
+			Platform struct {
+				Libvirt struct {
+					URI                    string `json:"URI"`
+					DefaultMachinePlatform struct {
+						Image string `json:"image"`
+					} `json:"defaultMachinePlatform"`
+					Network struct {
+						If      string `json:"if"`
+						IPRange string `json:"ipRange"`
+					} `json:"network"`
+					MasterIPs interface{} `json:"masterIPs"`
+				} `json:"libvirt"`
+			} `json:"platform"`
+			PullSecret string `json:"pullSecret"`
+		} `json:"config"`
+		File struct {
+			Filename string `json:"Filename"`
+			Data     string `json:"Data"`
+		} `json:"file"`
+	} `json:"*installconfig.InstallConfig"`
+	ICBaseDomain struct {
+		BaseDomain string `json:"BaseDomain"`
+	} `json:"*installconfig.baseDomain"`
+	ICClusterID struct {
+		ClusterID string `json:"ClusterID"`
+	} `json:"*installconfig.clusterID"`
+	ICClusterName struct {
+		ClusterName string `json:"ClusterName"`
+	} `json:"*installconfig.clusterName"`
+	ICPlatform struct {
+		Libvirt struct {
+			URI                    string `json:"URI"`
+			DefaultMachinePlatform struct {
+				Image string `json:"image"`
+			} `json:"defaultMachinePlatform"`
+			Network struct {
+				If      string `json:"if"`
+				IPRange string `json:"ipRange"`
+			} `json:"network"`
+			MasterIPs interface{} `json:"masterIPs"`
+		} `json:"libvirt"`
+	} `json:"*installconfig.platform"`
+	ICPullSecret struct {
+		PullSecret string `json:"PullSecret"`
+	} `json:"*installconfig.pullSecret"`
+	ICSSHPublicKey struct {
+		Key string `json:"Key"`
+	} `json:"*installconfig.sshPublicKey"`
+}
+
+// DataToValidate as expected base domain and cluster name
+type DataToValidate struct {
+	BaseDomain    string
+	ClusterName   string
+	SSH           string
+	PullSecret    string
+	ConnectionURI string
+}
+
+// ValidateInstallStateConfig check that the file content is the same of the env variables
+func ValidateInstallStateConfig(filePath string, expectedData DataToValidate) {
+	fileName := ".openshift_install_state.json"
+
+	openshiftInstState, err := os.Open(filePath)
+	if err != nil {
+		fmt.Println("Error opening the file:" + filePath)
+		log.Fatalf("Failed while open file: %v", err)
+	}
+
+	defer openshiftInstState.Close()
+	byteValue, _ := ioutil.ReadAll(openshiftInstState)
+
+	var installState installStateConf
+	json.Unmarshal(byteValue, &installState)
+
+	AssertStringContains(installState.InstallConfig.Config.BaseDomain,
```

Comment:
This and later parts of `ValidateInstallStateConfig` feel like a hand-crafted special case of [`reflect.DeepEqual`][1].  Is it doing something else?

[1]: https://golang.org/pkg/reflect/#DeepEqual